lib/lib.go

```diff
@@ -143,52 +142,29 @@ type FileConfig struct {
 // - $HOME/cfg/buildkite
 // - $HOME/.buildkite
 func LoadConfig(ctx context.Context) (*FileConfig, error) {
+
 	var filename string
 	var f *os.File
 	var err error
-	checkedLocations := make([]string, 0)
-	deadline, deadlineOk := ctx.Deadline()
-	if cfg, ok := os.LookupEnv("XDG_CONFIG_HOME"); ok {
-		filename = filepath.Join(cfg, "buildkite")
-		f, err = os.Open(filename)
-		checkedLocations = append(checkedLocations, filename)
-	}
+
+	filename, err = getCfgPath()
+	fmt.Println(filename)
 	if err != nil {
-		var homeDir string
-		homeDir, err = os.UserHomeDir()
-		if err != nil {
-			return nil, err
-		}
-		filename = filepath.Join(homeDir, "cfg", "buildkite")
-		f, err = os.Open(filename)
-		checkedLocations = append(checkedLocations, filename)
-		if err != nil { // fallback
-			rcFilename := filepath.Join(homeDir, ".buildkite")
-			f, err = os.Open(rcFilename)
-			checkedLocations = append(checkedLocations, rcFilename)
-		}
+		return nil, err
```

Comment:
sorry this diff is kind of crappy looking, i'm slightly concerned that this just dumps an unfriendly file error to the user if we hit this case, not sure how much you care about that?